engine/runtime-parser-processor-tests/src/main/java/org/enso/runtime/parser/processor/test/gen/ir/core/JName.java

```diff
@@ -0,0 +1,52 @@
+package org.enso.runtime.parser.processor.test.gen.ir.core;
+
+import org.enso.runtime.parser.dsl.IRChild;
+import org.enso.runtime.parser.dsl.IRNode;
+import org.enso.runtime.parser.processor.test.gen.ir.module.scope.JDefinition;
+import scala.collection.immutable.List;
+
+@IRNode
+public interface JName extends JExpression {
+  String name();
+
+  boolean isMethod();
+
+  @Override
+  JName duplicate(
+      boolean keepLocations,
+      boolean keepMetadata,
+      boolean keepDiagnostics,
+      boolean keepIdentifiers);
+
+  interface JBlank extends JName {
+    static JBlank create() {
+      return JNameGen.JBlankGen.builder().build();
```

Comment:
An alternative to generating **subclass** is to generate **superclass**. With such a style we could turn this `JBlank` interface into class. That way the Scala `case class` would be represented with Java `class` making it more 1:1 and simplify migration.

```java
package org.enso.runtime.parser.processor.test.gen.ir.core;

import org.enso.runtime.parser.dsl.IRChild;
import org.enso.runtime.parser.dsl.IRNode;
import org.enso.runtime.parser.processor.test.gen.ir.module.scope.JDefinition;
import scala.collection.immutable.List;

@IRNode
public interface Name extends Expression {
  String name();

  boolean isMethod();

  @Override
  JName duplicate(
      boolean keepLocations,
      boolean keepMetadata,
      boolean keepDiagnostics,
      boolean keepIdentifiers);

  @GenerateIR(interfaces="Name", superclass="Object")
  public final Blank extends BlankGen {
    public Blank(IdentifiedLocation identifiedLocation, MetadataStorage passData) {
      super(identifiedLocation, passData);
    }

    public String name() { return "_"; }

    // copy, duplicate, mapExpressions,
    // children, showCode will be generated into BlankGen

  }

  @GenerateIR(interfaces="Name", superclass="Object")
  public final Literal extends LiteralGen {
    public Literal(
      String name, boolean isMethod, IdentifiedLocation identifiedLocation, 
      @IRChild(required = false) Name originalName, MetadataStorage passData
    ) {
      super(name, isMethod, identifiedLocation, originalName, passData);
    }

    // copy, etc. &
    // originalName generated in the super class
  }

  @GenerateIR(interfaces="Name", superclass="Object")
  public final Qualified extends QualifiedGen {
    public Qualified(@IRChild List<JName> parts, IdentifiedLocation identifiedLocation, MetadataStorage passData) {
      super(parts, identifiedLocation, passData);
    }

    @Override
    public String name() {
      return parts().map(JName::name).mkString(".");
    }

    // children generated in superclass and return parts as annotated by @IRChild
  }
 }
```

Reply:
Super class approach pursued in https://github.com/enso-org/enso/pull/11770